_studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw.cpp

```diff
@@ -37,18 +54,23 @@ mfxStatus MFXVideoENCODEVP9_HW::Query(VideoCORE *core, mfxVideoParam *in, mfxVid
     }
     else
     {
+        eMFXHWType platform = core->GetHWType();
+
         // check attached buffers (all are supported by encoder, no duplications, etc.)
         mfxStatus sts = CheckExtBufferHeaders(in->NumExtParam, in->ExtParam);
         MFX_CHECK_STS(sts);
         sts = CheckExtBufferHeaders(out->NumExtParam, out->ExtParam);
         MFX_CHECK_STS(sts);
 
-        VP9MfxVideoParam toValidate = *in;
+        VP9MfxVideoParam toValidate(*in);
+        SetDefaultsForProfileAndFrameInfo(toValidate);
+        SetDefaultForLowpower(toValidate, platform);
 
         // get HW caps from driver
         ENCODE_CAPS_VP9 caps = {};
-        sts = QueryCaps(core, caps, GetGuid(toValidate), in->mfx.FrameInfo.Width, in->mfx.FrameInfo.Height);
-        MFX_CHECK(MFX_ERR_NONE == sts, MFX_ERR_UNSUPPORTED);
+        MFX_CHECK(MFX_ERR_NONE == QueryCaps(core, caps, GetGuid(toValidate), toValidate.mfx.FrameInfo.Width, toValidate.mfx.FrameInfo.Height), MFX_ERR_UNSUPPORTED);
```

Comment:
previous variant was better, try to not put functions call inside macro, it may cause accident double call during expansion